pkg/identity/revocation_db.go

```diff
@@ -22,25 +22,73 @@ type RevocationDB struct {
 	DB storage.KeyValueStore
 }
 
+// NewRevDB returns a new revocation database given the URL
+func NewRevDB(revocationDBURL string) (*RevocationDB, error) {
+	driver, source, err := dbutil.SplitConnstr(revocationDBURL)
+	if err != nil {
+		return nil, extensions.ErrRevocationDB.Wrap(err)
+	}
+
+	var db *RevocationDB
+	switch driver {
+	case "bolt":
+		db, err = NewRevocationDBBolt(source)
+		if err != nil {
+			return nil, extensions.ErrRevocationDB.Wrap(err)
+		}
+	case "redis":
+		db, err = NewRevocationDBRedis(revocationDBURL)
+		if err != nil {
+			return nil, extensions.ErrRevocationDB.Wrap(err)
+		}
+	default:
+		return nil, extensions.ErrRevocationDB.New("database scheme not supported: %s", driver)
+	}
+
+	return db, nil
+}
+
+// NewRevocationDBBolt creates a bolt-backed RevocationDB
+func NewRevocationDBBolt(path string) (*RevocationDB, error) {
```

Comment:
Do we need to expose the specific implementations?

Reply:
what do you mean?

Reply:
Ah, just whether `NewRevocationDBBolt` needs to be public or not?

Reply:
ah - good question.. i'll check.